src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/globalSelectionHeader.tsx

```diff
@@ -609,133 +331,144 @@ class GlobalSelectionHeader extends React.Component<Props, State> {
-                  projects={loadingProjects ? projects : memberProjects}
-                  loadingProjects={!initiallyLoaded && loadingProjects}
-                  nonMemberProjects={nonMemberProjects}
-                  value={this.state.projects || this.props.selection.projects}
-                  onChange={this.handleChangeProjects}
-                  onUpdate={this.handleUpdateProjects}
-                  multi={
-                    !disableMultipleProjectSelection && this.hasMultipleProjectSelection()
-                  }
-                  {...(loadingProjects ? paginatedProjectSelectorCallbacks : {})}
-                  showIssueStreamLink={showIssueStreamLink}
-                  showProjectSettingsLink={showProjectSettingsLink}
-                  lockedMessageSubject={lockedMessageSubject}
-                  footerMessage={projectsFooterMessage}
+                  defaultPeriod={defaultPeriod}
+                  hint={timeRangeHint}
                 />
-              );
-            }}
-          </Projects>
-        </HeaderItemPosition>
-
-        {showEnvironmentSelector && (
-          <React.Fragment>
-            <HeaderSeparator />
-            <HeaderItemPosition>
-              <MultipleEnvironmentSelector
-                organization={organization}
-                projects={this.props.projects}
-                loadingProjects={loadingProjects}
-                selectedProjects={selectedProjects}
-                value={this.state.environments || this.props.selection.environments}
-                onChange={this.handleChangeEnvironments}
-                onUpdate={this.handleUpdateEnvironmments}
-              />
-            </HeaderItemPosition>
-          </React.Fragment>
-        )}
-
-        {showDateSelector && (
-          <React.Fragment>
-            <HeaderSeparator />
-            <HeaderItemPosition>
-              <TimeRangeSelector
-                key={`period:${period}-start:${start}-end:${end}-utc:${utc}`}
-                showAbsolute={showAbsolute}
-                showRelative={showRelative}
-                relative={period}
-                start={start}
-                end={end}
-                utc={utc}
-                onChange={this.handleChangeTime}
-                onUpdate={this.handleUpdateTime}
-                organization={organization}
-                allowClearTimeRange={allowClearTimeRange}
-                hint={timeRangeHint}
-              />
-            </HeaderItemPosition>
-          </React.Fragment>
-        )}
-
-        {!showEnvironmentSelector && <HeaderItemPosition isSpacer />}
-        {!showDateSelector && <HeaderItemPosition isSpacer />}
-      </Header>
+              </HeaderItemPosition>
+            </React.Fragment>
+          )}
+
+          {!showEnvironmentSelector && <HeaderItemPosition isSpacer />}
+          {!showDateSelector && <HeaderItemPosition isSpacer />}
+        </Header>
+
+        {isGlobalSelectionReady ? children : <PageContent />}
```

Comment:
The header now blocks content from mounting until it is ready. Eg projects are loaded and single project is enforced if necessary. I’ve opted to not include a loader, but we could add one in here.